airflow/www/views.py

```diff
@@ -3237,7 +3237,8 @@ def duration(self, dag_id: str, session: Session = NEW_SESSION):
 
         # we must group any mapped TIs by dag_id, task_id, run_id
         mapped_tis = set()
-        tis_grouped = itertools.groupby(task_instances, lambda x: (x.dag_id, x.task_id, x.run_id))
+        tis_group_sort = lambda ti: (ti.dag_id, ti.task_id, ti.run_id)
```

Comment:
```suggestion

        def tis_group_sort(ti):
            return ti.dag_id, ti.task_id, ti.run_id

```

There’s a rule E731 that advises you to not assign a lambda but use def instead. We currently have it disabled, but it’s a good idea to follow it in new code.

Reply:
> There’s a rule E731 that advises you to not assign a lambda but use def instead.

Oh, I assume that we deliberately turned off E731. No problem I will wrap it into the generator

> With that said, TaskInstance actually has a function for this so this can just be operator.attrgetter("key").

Might be suitable for sorting, by not for grouping in this case